vital/tests/test_image_container_set.cxx

```diff
@@ -0,0 +1,68 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <test_common.h>
+
+
+#include <vital/types/image_container_set.h>
+#include <vital/types/image_container.h>
+
+namespace // anonymous
+{
+  // Helper methods for tests in this file
+}
+
+#define TEST_ARGS ()
+
+DECLARE_TEST_MAP();
+
+int
+main(int argc, char* argv[])
+{
+  CHECK_ARGS(1);
+
+  testname_t const testname = argv[1];
+
+  RUN_TEST(testname);
+}
+
+
+IMPLEMENT_TEST(simple_empty_container)
+{
+  kwiver::vital::image_container_set_sptr empty_set;
+  empty_set = std::make_shared<kwiver::vital::simple_image_container_set>();
+
+  // Check size is reported as zero
+  TEST_EQUAL("Set empty", empty_set->size(), 0);
+
+  // Check underlying vector implementation exists and also reports zero size
+  std::vector< kwiver::vital::image_container_sptr > data_ = empty_set->images();
+  TEST_EQUAL("Vector empty", data_.size(), 0);
```

Comment:
When porting to Google Test, I would highly encourage writing this without the temporary, i.e. `EXPECT_EQ(0, empty_set->images().size());`. This way, if the assertion fails, there is more information about the failure.